x-pack/platform/plugins/shared/dataset_quality/server/rule_types/fetch_es_query.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { IScopedClusterClient } from '@kbn/core/server';
+import { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/types';
+import type { estypes } from '@elastic/elasticsearch';
+import { extractKey } from './extract_key';
+
+const DEFAULT_GROUPS = 1000;
```

Comment:
Is this ok @LucaWintergerst?
We will be restricting the numbers of groups we query for creating the alerts up to `DEFAULT_GROUPS`.